Example project using Huub as a library

AllenZzw opened the pull request:
Add an example project using Huub as a library for solving jobshop scheduling problem

AllenZzw committed:
add an example package for jobshop scheduling

AllenZzw committed:
update

AllenZzw committed:
update

AllenZzw committed:
update engine to allow dynamic brancher

AllenZzw committed:
update example and expose trailedInt to external project

Dekker1 committed:
update example to current version of huub

Dekker1 committed:
refactor: split example into different modules

Dekker1 committed:
refactor: more descriptive brancher enum names

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/huub-solver/huub?pullRequest=244) <br/>All committers have signed the CLA.

Dekker1 committed:
docs: add missing docs

enable clippy for examples and fix these issues

Dekker1 committed:
refactor: update cli to use clap

Dekker1 committed:
style: sort rust items

Dekker1 commented:
I've refactored this some, and it seems in good shape to me now. The main thing that appears to be missing is a README file to tell people about the example and the format that it accepts (and maybe where to find some instance files). It would also be nice to have one or two created instances that we can keep in the repository and run as part of the test suite.

@AllenZzw Could you see whether you could complete this?

AllenZzw committed:
add readme and test instances

AllenZzw committed:
update solver statistics naming

AllenZzw committed:
update instances

AllenZzw commented:
> I've refactored this some, and it seems in good shape to me now. The main thing that appears to be missing is a README file to tell people about the example and the format that it accepts (and maybe where to find some instance files). It would also be nice to have one or two created instances that we can keep in the repository and run as part of the test suite.
> 
> @AllenZzw Could you see whether you could complete this?

I have updated the example with a README file and some example testing instances. @Dekker1 would be great if you could have a look before merging this.

Dekker1 committed:
fix: one sentence per line in md

Dekker1 committed:
style: sort test functions

Dekker1 committed:
ci: run the tests in examples
